### DISCUSSION TITLE
Auto Update Nextcloud AIO

### DISCUSSION THREAD

**Discussion by smooth-baking-powder** [nextcloud/all-in-one#3510] [Questions] [answered]
Hi,

may it possible to automate the Updates? For Example: The Updatecheck of the Mastercontainer was successful and an Update is available. On the Master Container you can set a "daily, Weekly (Sunday), monthly" Trigger with a Time "(00:00, 03:00, 12:00) for the Updateprocess. The Mastercontainer Start the upgrade automatically with this Trigger.

Or can i simply update all containers with Watchtower? So i need the settings to add the Label on all Containers.

Thanks

**Comment by szaimen** [comment, marked as answer]
Hi, this is already possible if you enable daily backups

**Reply by smooth-baking-powder** [reply]
Hey, do i need a big backup storage for this? for all Data? At the Moment i Skip this, cause i backup over vm hypervisor and Snapshots.

**Reply by szaimen** [reply]
Yes, all important files will be backed up using the built-in backup solution. If you dont want this see https://github.com/nextcloud/all-in-one#how-to-enable-automatic-updates-without-creating-a-backup-beforehand

**Reply by smooth-baking-powder** [reply]
Okay thanks. other question. it the cloud available while backup is in progress, if daily backup is enabled? i cant read information about this.

**Reply by szaimen** [reply]
> Okay thanks. other question. it the cloud available while backup is in progress, if daily backup is enabled? i cant read information about this.

It will stop the containers, do the backup and start them back up.

**Reply by smooth-baking-powder** [reply]
Okay :/ 3tb backup is to slow for offline backup... may it can be configured incremental?

**Reply by szaimen** [reply]
> Okay :/ 3tb backup is to slow for offline backup... may it can be configured incremental?

AIOs backup solution is incremental, differential, compressed and encrypted. We are using borg for that: https://borgbackup.readthedocs.io/en/stable/index.html 

So only the first backup will take long. Afterwards only changes will be taken into account.

**Reply by smooth-baking-powder** [reply]
Ok, that sounds great. i will test it. Thank you for the details and informations 👍

**Reply by szaimen** [reply]
I just made the readme a bit more verbose about what borg is able to do: https://github.com/nextcloud/all-in-one/commit/b5577a2909965d28e6087e685876b4ae0149d181

**Reply by szaimen** [reply]
another hint: https://github.com/nextcloud/all-in-one/commit/1f68615c8d18181a4b053247aaea785cee15334b

**Reply by lowChecker** [reply]
@szaimen Thanks for doing a great job with NextCould AIO and using the borg backup which worked well for me. But I changed my setup to Proxmox Backup Server. I backup the docker host virtual machine to the Proxmox Backup Server and deativated the AIO borg backup to safe space. 
So, I do have a proper backup solution with daily backups. But that is outside of the NextCloud AIO universe. 
Is there a hack to enable automatic update without borg?  Or can I use ansible to start "manualy" the update process?

**Reply by szaimen** [reply]
https://github.com/nextcloud/all-in-one?tab=readme-ov-file#how-to-enable-automatic-updates-without-creating-a-backup-beforehand

**Comment by Wadera** [comment]
What if I'm doing backup on storage level for whole VMs and I want only trigger updates? 

Should I just run these commands using ansible / cron?
https://github.com/nextcloud/all-in-one/discussions/3008 

Or there is better option?

**Reply by szaimen** [reply]
See https://github.com/nextcloud/all-in-one?tab=readme-ov-file#how-to-enable-automatic-updates-without-creating-a-backup-beforehand

**Reply by Wadera** [reply]
That's perfect. I don't know how I missed that. Thank you!

**Reply by Sturmey66** [reply]
There should also be a warning if you try to update using AIO and you haven't updated in a month. When users aren't updating daily, the AIO update process fails. It would be good to add some type of warning during the installation that daily updates are recommended and where to find the steps.

**Reply by szaimen** [reply]
See https://github.com/nextcloud/all-in-one?tab=readme-ov-file#how-often-are-update-notifications-sent
